modules/cache-common/src/main/java/org/opensearch/cache/common/tier/TieredSpilloverCacheStatsHolder.java

```diff
@@ -0,0 +1,158 @@
+            : "Invalid tier dimension value";
+        return tierDimensionValue;
+    }
+
+    private boolean isLeafNode(int depth) {
+        return depth == dimensionNames.size(); // Not size - 1, because there is also a root node
+    }
+
+    @Override
+    public void incrementHits(List<String> dimensionValues) {
+        getAndCheckTierDimensionValue(dimensionValues);
+        // Hits from either tier should be included in the total values.
+        internalIncrement(dimensionValues, (node, depth) -> node.incrementHits(), true);
+    }
+
+    @Override
+    public void incrementMisses(List<String> dimensionValues) {
+        final String tierValue = getAndCheckTierDimensionValue(dimensionValues);
+
+        // If the disk tier is present, only misses from the disk tier should be included in total values.
+        BiConsumer<Node, Integer> missIncrementer = (node, depth) -> {
+            if (tierValue.equals(TIER_DIMENSION_VALUE_ON_HEAP) && diskCacheEnabled) {
+                // If on-heap tier, increment only the leaf node corresponding to the on heap values; not the total values in its parent
+                // nodes
+                if (isLeafNode(depth)) {
+                    node.incrementMisses();
+                }
+            } else {
+                // If disk tier, or on-heap tier with a disabled disk tier, increment the leaf node and its parents
+                node.incrementMisses();
+            }
+        };
+
+        internalIncrement(dimensionValues, missIncrementer, true);
+    }
+
+    @Override
+    public void incrementEvictions(List<String> dimensionValues) {
+        final String tierValue = getAndCheckTierDimensionValue(dimensionValues);
+
+        // If the disk tier is present, only evictions from the disk tier should be included in total values.
+        BiConsumer<DefaultCacheStatsHolder.Node, Integer> evictionsIncrementer = (node, depth) -> {
+            if (tierValue.equals(TIER_DIMENSION_VALUE_ON_HEAP) && diskCacheEnabled) {
+                // If on-heap tier, increment only the leaf node corresponding to the on heap values; not the total values in its parent
+                // nodes
+                if (isLeafNode(depth)) {
+                    node.incrementEvictions();
+                }
+            } else {
+                // If disk tier, or on-heap tier with a disabled disk tier, increment the leaf node and its parents
+                node.incrementEvictions();
+            }
+        };
+
+        internalIncrement(dimensionValues, evictionsIncrementer, true);
+    }
+
+    @Override
+    public void incrementSizeInBytes(List<String> dimensionValues, long amountBytes) {
+        getAndCheckTierDimensionValue(dimensionValues);
+        // Size from either tier should be included in the total values.
+        internalIncrement(dimensionValues, (node, depth) -> node.incrementSizeInBytes(amountBytes), true);
```

Comment:
We should delegate this invocation to the overridden method

Reply:
Or even better is for the overridden methods to expose a callback. And this class can override the callback to implement the validation logic. For example in base class:

```
protected void validationCallback() {
    // does nothing in base class
}

private void incrementSizeinBytes(List<String> dimensionValues, long amountBytes) {
    validationCallback(dimensionValues);
    // logic
}
```

In super class here:

```
@Override
private void validationCallback() {
        String tierDimensionValue = dimensionValues.get(dimensionValues.size() - 1);
        assert tierDimensionValue.equals(TIER_DIMENSION_VALUE_ON_HEAP) || tierDimensionValue.equals(TIER_DIMENSION_VALUE_DISK)
            : "Invalid tier dimension value";
        return tierDimensionValue;
}
```